tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Catalogue Solutions Update Validation.feature

```diff
@@ -0,0 +1,62 @@
+
+Background:
+    Given Orders exist
+        | OrderId    | Description      | LastUpdatedBy                        | LastUpdatedByName | OrganisationId                       | CommencementDate |
+        | C000014-01 | Some Description | 335392e4-4bb1-413b-9de5-36a85c9c0422 | Tom Smith         | 4af62b99-638c-4247-875e-965239cd0c48 | 01/01/2021       |
+    And Service Recipients exist
+        | OdsCode | Name                | OrderId    |
+        | ODS1    | Service Recipient   | C000014-01 |
+    And Order items exist
+        | OrderId    | OdsCode | CatalogueItemName | CatalogueItemType | ProvisioningType |
+        | C000014-01 | ODS1    | Order Item 1      | Solution          | OnDemand         |
+    And the user is logged in with the Buyer role for organisation 4af62b99-638c-4247-875e-965239cd0c48
+
+@7840
+Scenario: 1. Update a catalogue solution order item with missing values
+    Given the user creates a request to change the catalogue solution order item ('Order Item 1') for the order with ID 'C000014-01'
+    And the user enters the '<payload-type>' update catalogue solution order item request payload
+    When the user sends the update catalogue solution order item request
+    Then a response with status code 400 is returned
+    And the response contains the following errors
+        | id         | field         |
+        | <error-id> | <error-field> |
+
+    Examples: Request payloads
+        | payload-type                    | error-id                                   | error-field           |
+        | missing-delivery-date           | DeliveryDateRequired                       | DeliveryDate          |
+        | missing-quantity                | QuantityRequired                           | Quantity              |
+        | missing-estimation-period       | EstimationPeriodRequiredIfVariableOnDemand | EstimationPeriod      |
+        | missing-price                   | PriceRequired                              | Price                 |
+
+@7840
+Scenario: 2. Create catalogue solution order item with out of range values
+    Given the user creates a request to change the catalogue solution order item ('Order Item 1') for the order with ID 'C000014-01'
+    And the user enters the '<payload-type>' update catalogue solution order item request payload
+    When the user sends the update catalogue solution order item request
+    Then a response with status code 400 is returned
+    And the response contains the following errors
+        | id         | field         |
+        | <error-id> | <error-field> |
+
+    Examples: Request payloads
+        | payload-type               | error-id                          | error-field  |
+        | outside-delivery-window    | DeliveryDateOutsideDeliveryWindow | DeliveryDate |
+        | greater-than-zero-quantity | QuantityGreaterThanZero           | Quantity     |
+
+@7840
+Scenario: 3. Create catalogue solution order item with invalid values
```

Comment:
Can we please get some tests for all the different fields being empty / invalid, e.g. wrong provisioning type & making sure the api doesn't throw 500 errors?